+sw_tests/+unit_tests/unittest_spinw_addatom.m

```diff
@@ -0,0 +1,185 @@
+        swobj = [];
+        abstol = 10*eps
+        default_unit_cell = struct('r', [0; 0; 0], 'S', 0, ...
+            'label', {{'atom_1'}}, 'color', int32([255; 0; 0]), 'ox', 0, ...
+            'occ', 1, 'b', [1; 1], 'ff', [zeros(2,10) [1; 1]], ...
+            'A', int32(-1), 'Z', int32(113), 'biso', 0)
+    end
+    properties (TestParameter)
+        property_error = {{'Z', 'spinw:addatom:WrongInput'}, ...
+            {'A', 'spinw:addatom:WrongInput'}, ...
+            {'biso', 'spinw:addatom:WrongInput'}, ...
+            {'ox', 'spinw:addatom:WrongInput'}, ...
+            {'S', 'spinw:sw_valid:SizeMismatch'}};
+        oxidation = {3, -3};
+        pos_vector = {[0;0;0], [0 0 0]}
+    end
+
+    methods (TestMethodSetup)
+        function setup_spinw_model(testCase)
+            testCase.swobj = spinw(); % default init
+        end
+    end
+
+    methods (Test)
+        function test_no_input_calls_help(testCase)
+            % Tests that if call addmatrix with no input, it calls the help
+            help_function = sw_tests.utilities.mock_function('swhelp');
+            testCase.swobj.addatom();
+            testCase.assertEqual(help_function.n_calls, 1);
+            testCase.assertEqual(help_function.arguments, ...
+                {{'spinw.addatom'}});
+        end
+               
+        function test_add_multiple_atom_throws_mismatch_parameter_size(testCase, property_error)
+            [prop, error] = property_error{:};
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0 0;0 0.5;0 0.5], prop, 2), ...
+                error)
+        end
+        
+        function test_add_atom_fails_invalid_position_size(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0, 0]), ...
+                'MATLAB:NonIntegerInput')  % need a better error for this
+        end
+        
+        function test_add_atom_fails_without_position(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('S', 1), ...
+                'sw_readparam:MissingParameter')
+        end
+        
+        function test_add_atom_fails_with_negative_spin(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0; 0; 0], 'S', -1), ...
+                'spinw:addatom:WrongInput')
+        end
+        
+        function test_add_atom_warns_bx_provided(testCase)
+            testCase.verifyWarning(...
+                @() testCase.swobj.addatom('r', [0; 0; 0], 'bx', 2), ...
+                'spinw:addatom:WrongInput')
+        end
+        
+        function test_add_atom_warns_b_and_bn_provided(testCase)
+            testCase.verifyWarning(...
+                @() testCase.swobj.addatom('r', [0; 0; 0], 'b', 2, 'bn', 3), ...
+                'spinw:addatom:WrongInput')
+            % check value for scattering length is bn provided
+            testCase.assertEqual(testCase.swobj.unit_cell.b(1,1), 3)
```

Comment:
It would be good to have a test if only `'b'` is provided too, as the logic is a bit strange here I worry we could lose it if we're not careful!